server/src/test/java/org/elasticsearch/gateway/GatewayIndexStateIT.java

```diff
@@ -550,90 +549,6 @@ public void testHalfDeletedIndexImport() throws Exception {
         assertBusy(() -> assertThat(internalCluster().getInstance(NodeEnvironment.class).availableIndexFolders(), empty()));
     }
 
-    public void testOnlyWritesIndexMetaDataFilesOnDataNodes() throws Exception {
```

Comment:
I think this test has value, although it needs adjusting and maybe moving somewhere more appropriate. We do want to check that these folders are cleaned up when indices are de-allocated and/or deleted.

Reply:
There is an equivalent test in MetaDataWriteDataNodesIT that I've extended a bit in cc64a19.